sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -625,6 +637,12 @@ Exceptions:
   A cycle may be introduced to the graph via a call to `make_edge()` that
   creates a forward dependency.
 
+* Throws synchronously with error code `invalid` if the graph contains a
+  node which targets a device not present in `syclContext`.
+
+* Throws synchronously with error code `invalid` if more than one device is
+  targeted by the nodes of the graph.
```

Comment:
I've put the error restricting the graph to being a single device on `finalize()`, however I know folks like errors to be thrown as early as possible to help the user, so instead we could put it on `graph.add()` for explicit API and add some error wording to record and replay.

Reply:
I like throwing an error from `graph.add()` , should be easily detectable.

Reply:
Moved this error from `graph.finalize` to  `graph.add()` and `graph.begin_recording()`.

But I also added some error wording in the section on record & replay error handling, because with the error just specified on `graph.begin_recording()` it possible for a user to work around that with the following code

```cpp
graph.begin_recording(queue_device_A);
graph.begin_recording(queue_device_B);  // We've not added any nodes to the graph yet, so no specified error thrown


queue_device_A.submit();  // Add a node targetting device A
queue_device_B.submit();  // Try to add a node targetting device B, throw an error now
```